### ISSUE TITLE
First eshell prompt is not displayed in colour, subsequent prompts are

### ISSUE THREAD

**Issue by jen20** [syl20bnr/spacemacs#5542]
#### Description
The first time I open an eshell window, the prompt does not appear in colour, regardless of whether the prompt is set using `eshell-prompt-function` or left as default. Pressing enter at the prompt then resumes colour printing of the prompt (again for either the default or custom prompts).

#### Reproduction guide
- Start Emacs
- Start an eshell using `<SPC '>
- Observe the prompt is not displayed in colour
- Press `<ENTER>` at the prompt
- Observe the new prompt displayed is in colour.

*Observed behaviour:*
First prompt displayed does not follow the colour scheme of subsequent prompts and is displayed in monochrome.

*Expected behaviour:*
First prompt displays the same colour scheme as subsequent prompts.

#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.13
- Spacemacs branch: master (rev. 2c3e6b4)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(osx auto-completion better-defaults c-c   csharp dash dockerfile emacs-lisp erlang git github go html javascript lua markdown react ruby-on-rails
     (shell :variables shell-default-shell 'eshell shell-default-height 30 shell
-default-position 'bottom)
     spell-checking syntax-checking terraform version-control windows-scripts yaml)
```

**Comment by jen20** [comment]
An additional note - I just noticed that there were updates to spacemacs available on `master`, so I am now on commit `a298a5a`, where the same behaviour is exhibited.

**Comment by TheBB** [comment]
I can repro this by the way. Annoys me too.

**Comment by TheBB** [comment]
The problem is that we apply xterm coloring to eshell output, but the setting `xterm-color-preserve-properties` is only set to `t` in `eshell-mode-hook` which is *AFTER* the first prompt has been displayed.

**Comment by TheBB** [comment]
Fixed in #5703.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
